react/src/component/app.js

```diff
@@ -0,0 +1,94 @@
+import { useState, useEffect, useContext } from "react";
+import {
+    ContextAppName,
+    ContextSimulationListPromise,
+    ContextSchema,
+    ContextLayouts
+} from "../context"
+import { configureStore } from "@reduxjs/toolkit"
+import { modelsSlice } from "../store/models";
+import { formStatesSlice } from "../store/formState";
+import { useSetup } from "../hook/setup";
+import { compileSchemaFromJson } from "../utility/schema";
+import { Provider } from "react-redux";
+import { SimulationBrowserRoot } from "../component/simbrowser";
+import { Layouts } from "../layout/layouts";
+import "./app.scss";
+
+function SimulationListInitializer(props) {
+    let stateFn = useState;
```

Comment:
These aliases are just to make React's linter happy. They don't like it when we access hooks from 'uncertain' code paths. We base things off of the schema which is assumed to be constant so even some conditionals can be considered to be consistent across all renders if they are just asking about the schema. The aliases aren't needed in all the places they are present. I need to go around and check them again. I will also look into ignoring the linter checks but they may help our other developers get used to React

Reply:
I think it is ok to use something like "linterUseState" or something like that. Could be something shorter that's unique. I assume it can't be a suffix e.g. userStateRLH (react linter happy).